File: scripts/test-common-foods.mjs
/**
 * Built-in food catalog tests — run: node scripts/test-common-foods.mjs
 */
import { searchCommonFoods, getCommonFoodById } from '../src/utils/commonFoodSearch.js'
import { scaleNutrients } from '../src/utils/nutrients.js'

let failed = 0

function assert(cond, msg) {
  if (!cond) {
    console.error(`✗ ${msg}`)
    failed += 1
  } else {
    console.log(`✓ ${msg}`)
  }
}

const chickenHits = searchCommonFoods('chicken breast')
assert(chickenHits[0]?.id === 'chicken_breast_cooked', 'finds chicken breast')

const riceHits = searchCommonFoods('rice')
assert(riceHits.some((f) => f.id === 'white_rice_cooked'), 'finds white rice')
assert(riceHits.some((f) => f.id === 'rice_cakes_plain'), 'finds rice cakes')

const beef = getCommonFoodById('ground_beef_93_lean_cooked')
const scaled = scaleNutrients(beef.nutrients_per_100g, 150)
assert(Math.abs(scaled.protein_g - 41.1) < 0.01, '150g 93% beef protein scales correctly')

const chicken = getCommonFoodById('chicken_breast_cooked')
assert(chicken.nutrients_per_100g.protein_g === 31, 'per 100g protein preserved')
const portion = scaleNutrients(chicken.nutrients_per_100g, 200)
assert(portion.protein_g === 62, '200g chicken = 62g protein')

if (failed > 0) process.exit(1)
console.log('All common food tests passed')

File: athletyx/backend/tests/test_common_foods.py
"""Tests for built-in common food catalog."""

from backend.common_foods import get_common_food, search_common_foods
from backend.nutrition_service import scale_nutrients


def test_search_chicken_breast():
    results = search_common_foods("chicken breast")
    assert results
    assert results[0]["catalog_id"] == "chicken_breast_cooked"
    assert results[0]["nutrients_per_100g"]["protein_g"] == 31


def test_search_rice_cakes():
    results = search_common_foods("rice cakes")
    assert any(r["catalog_id"] == "rice_cakes_plain" for r in results)


def test_search_ground_beef():
    results = search_common_foods("ground beef")
    assert len(results) >= 2


def test_get_common_food_scaling():
    food = get_common_food("chicken_breast_cooked")
    assert food is not None
    scaled = scale_nutrients(food["nutrients_per_100g"], 200)
    assert scaled["protein_g"] == 62
    assert scaled["calories_kcal"] == 330
